src/test/java/com/google/opensesame/projects/ProjectEntityTest.java

```diff
@@ -0,0 +1,161 @@
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.googlecode.objectify.ObjectifyFactory;
+import com.googlecode.objectify.ObjectifyService;
+import com.googlecode.objectify.util.Closeable;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.Collection;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ProjectEntityTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private Closeable session;
+
+  private ProjectEntity firstMockProject;
+  private ProjectEntity secondMockProject;
+  private ProjectEntity thirdMockProject;
+
+  private HttpServletResponse mockResponse;
+  private StringWriter responseStringWriter;
+
+  @BeforeClass
+  public static void objectifyClassSetup() {
+    ObjectifyService.setFactory(new ObjectifyFactory());
+    ObjectifyService.register(ProjectEntity.class);
+  }
+
+  @Before
+  public void objectifyStartSession() throws IOException {
+    session = ObjectifyService.begin();
+    helper.setUp();
+  }
+
+  @Before
+  public void mockProjectSetUp() {
+    firstMockProject =
+        new ProjectEntity(
+            "273537467", Arrays.asList("Mentor 1", "Mentor 2", "Mentor 3"), Arrays.asList());
+    secondMockProject =
+        new ProjectEntity("45717250", Arrays.asList("Mentor 1", "Mentor 2"), Arrays.asList());
+    thirdMockProject = new ProjectEntity("20580498", Arrays.asList("Mentor 1"), Arrays.asList());
+    ofy().save().entities(firstMockProject, secondMockProject, thirdMockProject).now();
+  }
+
+  @Before
+  public void mockResponseSetUp() throws IOException {
+    mockResponse = mock(HttpServletResponse.class);
+    responseStringWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseStringWriter));
+  }
+
+  @After
+  public void objectifyEndSession() {
+    session.close();
+    helper.tearDown();
+  }
+
+  @Test
+  public void queryByIds() throws IOException {
+    // Expect to receive the ProjectEntities with the corresponding repository IDs.
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    when(request.getParameterValues(ProjectEntity.PROJECT_ID_PARAM))
+        .thenReturn(new String[] {firstMockProject.repositoryId, secondMockProject.repositoryId});
+
+    Collection<ProjectEntity> queryResult = ProjectEntity.queryFromRequest(request, mockResponse);
+    Collection<ProjectEntity> expected = Arrays.asList(firstMockProject, secondMockProject);
+    assertNotNull(queryResult);
```

Comment:
```suggestion

    assertNotNull(queryResult);
```
A test can generally follow the pattern "setup" "perform action under test" and "check correctness" and I like to structure it that way to make it easy to read. Here it's mostly good, I would just add a newline after performing the "action under test" before checking correctness.

Reply:
Thanks for the tips. I agree with the suggestion and will apply it!